<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>AddGradesDialog</class>
 <widget class="QDialog" name="AddGradesDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>505</width>
    <height>444</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>0</width>
    <height>0</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>780</width>
    <height>580</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Add Grades</string>
  </property>
  <property name="styleSheet">
   <string notr="true">background-color: #f5f7fa;</string>
  </property>
  <layout class="QVBoxLayout" name="mainLayout">
   <property name="spacing">
    <number>0</number>
   </property>
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item>
    <widget class="QFrame" name="headerFrame">
     <property name="minimumSize">
      <size>
       <width>0</width>
       <height>80</height>
      </size>
     </property>
     <property name="maximumSize">
      <size>
       <width>16777215</width>
       <height>80</height>
      </size>
     </property>
     <property name="styleSheet">
      <string notr="true">background: qlineargradient(x1:0, y1:0, x2:1, y2:1, stop:0 #f093fb, stop:1 #f5576c); border-radius: 0px;</string>
     </property>
     <property name="frameShape">
      <enum>QFrame::Shape::StyledPanel</enum>
     </property>
     <layout class="QHBoxLayout" name="headerLayout">
      <property name="leftMargin">
       <number>30</number>
      </property>
      <property name="topMargin">
       <number>20</number>
      </property>
      <property name="rightMargin">
       <number>30</number>
      </property>
      <property name="bottomMargin">
       <number>20</number>
      </property>
      <item>
       <widget class="QLabel" name="labelTitle">
        <property name="styleSheet">
         <string notr="true">font-size: 24px; font-weight: bold; color: white; background: transparent;</string>
        </property>
        <property name="text">
         <string>üìù  Add Grades</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="headerSpacer">
        <property name="orientation">
         <enum>Qt::Orientation::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QScrollArea" name="scrollArea">
     <property name="styleSheet">
      <string notr="true">border: none; background-color: #f5f7fa;</string>
     </property>
     <property name="widgetResizable">
      <bool>true</bool>
     </property>
     <widget class="QWidget" name="scrollContent">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>505</width>
        <height>284</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: #f5f7fa;</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_3">
       <property name="leftMargin">
        <number>15</number>
       </property>
       <property name="topMargin">
        <number>10</number>
       </property>
       <property name="rightMargin">
        <number>15</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <property name="horizontalSpacing">
        <number>15</number>
       </property>
       <property name="verticalSpacing">
        <number>30</number>
       </property>
       <item row="0" column="0">
        <layout class="QVBoxLayout" name="verticalLayout">
         <property name="spacing">
          <number>0</number>
         </property>
         <item>
          <widget class="QLabel" name="labelStudentID">
           <property name="styleSheet">
            <string notr="true">font-size: 13px; color: #718096; font-weight: bold; background: transparent;</string>
           </property>
           <property name="text">
            <string>Student ID *</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="comboBoxStudentID">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>150</width>
             <height>40</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>16777215</width>
             <height>65</height>
            </size>
           </property>
           <property name="styleSheet">
            <string notr="true">/* ----------------- QComboBox Base ----------------- */
QComboBox {
    border: 2px solid #E0E0E0;          /* main border */
    border-radius: 10px;
    padding: 8px 10px;                   /* Matches QLineEdit padding */
    background: rgba(250, 250, 250, 200);
    font-size: 11pt;
    color: #111111;
}

/* Focused state */
QComboBox:focus {
    border: 2px solid #667EEA; 
    background: rgba(250, 250, 250, 220);
}

/* ----------------- Drop-down Arrow ----------------- */
QComboBox::drop-down {
    subcontrol-origin: padding;
    subcontrol-position: top right;
    width: 30px; 
    border-left-width: 1px;
    border-left-color: #E0E0E0;
    border-left-style: solid;
    border-top-right-radius: 12px;       /* rounder corners */
    border-bottom-right-radius: 12px;    /* rounder corners */
    background: transparent;
}

/* Remove ugly black border on hover/focus */
QComboBox::drop-down:hover,
QComboBox::drop-down:focus {
    border-left-color: #667EEA;
    background: transparent;
    border: none;
}

/* Modern, visible arrow using simple chevron */
QComboBox::down-arrow {
    width: 10px;
    height: 10px;
    border: solid #667EEA;
    border-width: 0 2px 2px 0;
    margin-right: 6px;
}

/* ----------------- Dropdown Popup ----------------- */
/* Dropdown Popup */
QComboBox QAbstractItemView {
    background-color: #fdfdfd;          /* slightly off-white */
    border: 2px solid #E0E0E0;          /* subtle grey border */
    border-radius: 8px;                  /* may not be perfect on Windows */
    color: #111111;
    font-size: 11pt;
    padding: 5px;
    selection-background-color: #667EEA;
    selection-color: white;
    outline: none;
}

/* Items hover/selection */
QComboBox QAbstractItemView::item:hover {
    background-color: rgba(102, 126, 234, 0.1);
    color: #111111;
}

QComboBox QAbstractItemView::item:selected {
    background-color: #667EEA;
    color: white;
}

/* Error highlight for combo box */
QComboBox[error=&quot;true&quot;] {
    border: 2px solid #FF4A4A;
}</string>
           </property>
           <property name="editable">
            <bool>true</bool>
           </property>
           <property name="placeholderText">
            <string>ID...</string>
           </property>
           <item>
            <property name="text">
             <string>Computer</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Communication</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Power</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Biomedical</string>
            </property>
           </item>
          </widget>
         </item>
        </layout>
       </item>
       <item row="0" column="1">
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <property name="spacing">
          <number>0</number>
         </property>
         <item>
          <widget class="QLabel" name="labelCourseCode">
           <property name="styleSheet">
            <string notr="true">font-size: 13px; color: #718096; font-weight: bold; background: transparent;</string>
           </property>
           <property name="text">
            <string>Course Code *</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="comboBoxSelectCourse">
           <property name="minimumSize">
            <size>
             <width>200</width>
             <height>45</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>16777215</width>
             <height>65</height>
            </size>
           </property>
           <property name="cursor">
            <cursorShape>PointingHandCursor</cursorShape>
           </property>
           <property name="styleSheet">
            <string notr="true">/* ----------------- QComboBox Base ----------------- */
QComboBox {
    border: 2px solid #E0E0E0;          /* main border */
    border-radius: 10px;
    padding: 8px 10px;                   /* Matches QLineEdit padding */
    background: rgba(250, 250, 250, 200);
    font-size: 11pt;
    color: #111111;
}

/* Focused state */
QComboBox:focus {
    border: 2px solid #667EEA; 
    background: rgba(250, 250, 250, 220);
}

/* ----------------- Drop-down Arrow ----------------- */
QComboBox::drop-down {
    subcontrol-origin: padding;
    subcontrol-position: top right;
    width: 30px; 
    border-left-width: 1px;
    border-left-color: #E0E0E0;
    border-left-style: solid;
    border-top-right-radius: 12px;       /* rounder corners */
    border-bottom-right-radius: 12px;    /* rounder corners */
    background: transparent;
}

/* Remove ugly black border on hover/focus */
QComboBox::drop-down:hover,
QComboBox::drop-down:focus {
    border-left-color: #667EEA;
    background: transparent;
    border: none;
}

/* Modern, visible arrow using simple chevron */
QComboBox::down-arrow {
    width: 10px;
    height: 10px;
    border: solid #667EEA;
    border-width: 0 2px 2px 0;
    margin-right: 6px;
}

/* ----------------- Dropdown Popup ----------------- */
/* Dropdown Popup */
QComboBox QAbstractItemView {
    background-color: #fdfdfd;          /* slightly off-white */
    border: 2px solid #E0E0E0;          /* subtle grey border */
    border-radius: 8px;                  /* may not be perfect on Windows */
    color: #111111;
    font-size: 11pt;
    padding: 5px;
    selection-background-color: #667EEA;
    selection-color: white;
    outline: none;
}

/* Items hover/selection */
QComboBox QAbstractItemView::item:hover {
    background-color: rgba(102, 126, 234, 0.1);
    color: #111111;
}

QComboBox QAbstractItemView::item:selected {
    background-color: #667EEA;
    color: white;
}

/* Error highlight for combo box */
QComboBox[error=&quot;true&quot;] {
    border: 2px solid #FF4A4A;
}</string>
           </property>
           <property name="editable">
            <bool>true</bool>
           </property>
           <item>
            <property name="text">
             <string>Select a course...</string>
            </property>
           </item>
          </widget>
         </item>
        </layout>
       </item>
       <item row="1" column="0">
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <property name="spacing">
          <number>0</number>
         </property>
         <item>
          <widget class="QLabel" name="labelCreditHours">
           <property name="styleSheet">
            <string notr="true">font-size: 13px; color: #718096; font-weight: bold; background: transparent;</string>
           </property>
           <property name="text">
            <string>Grade *</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="comboBoxGrade">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>100</width>
             <height>40</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>16777215</width>
             <height>65</height>
            </size>
           </property>
           <property name="styleSheet">
            <string notr="true">/* ----------------- QComboBox Base ----------------- */
QComboBox {
    border: 2px solid #E0E0E0;          /* main border */
    border-radius: 10px;
    padding: 8px 10px;                   /* Matches QLineEdit padding */
    background: rgba(250, 250, 250, 200);
    font-size: 11pt;
    color: #111111;
}

/* Focused state */
QComboBox:focus {
    border: 2px solid #667EEA; 
    background: rgba(250, 250, 250, 220);
}

/* ----------------- Drop-down Arrow ----------------- */
QComboBox::drop-down {
    subcontrol-origin: padding;
    subcontrol-position: top right;
    width: 30px; 
    border-left-width: 1px;
    border-left-color: #E0E0E0;
    border-left-style: solid;
    border-top-right-radius: 12px;       /* rounder corners */
    border-bottom-right-radius: 12px;    /* rounder corners */
    background: transparent;
}

/* Remove ugly black border on hover/focus */
QComboBox::drop-down:hover,
QComboBox::drop-down:focus {
    border-left-color: #667EEA;
    background: transparent;
    border: none;
}

/* Modern, visible arrow using simple chevron */
QComboBox::down-arrow {
    width: 10px;
    height: 10px;
    border: solid #667EEA;
    border-width: 0 2px 2px 0;
    margin-right: 6px;
}

/* ----------------- Dropdown Popup ----------------- */
/* Dropdown Popup */
QComboBox QAbstractItemView {
    background-color: #fdfdfd;          /* slightly off-white */
    border: 2px solid #E0E0E0;          /* subtle grey border */
    border-radius: 8px;                  /* may not be perfect on Windows */
    color: #111111;
    font-size: 11pt;
    padding: 5px;
    selection-background-color: #667EEA;
    selection-color: white;
    outline: none;
}

/* Items hover/selection */
QComboBox QAbstractItemView::item:hover {
    background-color: rgba(102, 126, 234, 0.1);
    color: #111111;
}

QComboBox QAbstractItemView::item:selected {
    background-color: #667EEA;
    color: white;
}

/* Error highlight for combo box */
QComboBox[error=&quot;true&quot;] {
    border: 2px solid #FF4A4A;
}</string>
           </property>
           <property name="placeholderText">
            <string>Select Grade...</string>
           </property>
           <item>
            <property name="text">
             <string>A+</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>A</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>B+</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>B</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>C+</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>C</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>D+</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>D</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>F</string>
            </property>
           </item>
          </widget>
         </item>
        </layout>
       </item>
       <item row="1" column="1">
        <layout class="QGridLayout" name="gridLayout">
         <item row="0" column="0">
          <widget class="QLabel" name="labelSemesterYear">
           <property name="styleSheet">
            <string notr="true">font-size: 13px; color: #718096; font-weight: bold; background: transparent;</string>
           </property>
           <property name="text">
            <string>Semester year *</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QSpinBox" name="spinBoxSemesterYear">
           <property name="minimumSize">
            <size>
             <width>100</width>
             <height>40</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>16777215</width>
             <height>65</height>
            </size>
           </property>
           <property name="styleSheet">
            <string notr="true">QSpinBox {
    border: 2px solid #E0E0E0;
    border-radius: 10px;
    padding: 8px 10px;
    background: rgba(250, 250, 250, 200);
    font-size: 11pt;
    color: #111111;
}

/* Focus border (same as QLineEdit) */
QSpinBox:focus {
    border: 2px solid #667EEA;
    background: rgba(250, 250, 250, 220);
}

/* Internal text area matches QLineEdit text style */
QSpinBox QLineEdit {
    border: none;
    background: transparent;
    font-size: 11pt;
    color: #111111;
}

/* Placeholder-like behavior when disabled */
QSpinBox:disabled,
QSpinBox QLineEdit:disabled {
    background: #ededed;
    color: #a1a8b3;
}

/* Hide up/down buttons (matches your modern clean UI) */
QSpinBox::up-button,
QSpinBox::down-button {
    width: 0;
    height: 0;
    border: none;
}
</string>
           </property>
           <property name="buttonSymbols">
            <enum>QAbstractSpinBox::ButtonSymbols::NoButtons</enum>
           </property>
           <property name="accelerated">
            <bool>false</bool>
           </property>
           <property name="maximum">
            <number>3000</number>
           </property>
           <property name="value">
            <number>2025</number>
           </property>
           <property name="displayIntegerBase">
            <number>10</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="2" column="0">
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Orientation::Vertical</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Policy::Minimum</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>17</width>
           <height>30</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
      <widget class="QPushButton" name="pushButton">
       <property name="geometry">
        <rect>
         <x>198</x>
         <y>169</y>
         <width>62</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>PushButton</string>
       </property>
      </widget>
     </widget>
    </widget>
   </item>
   <item>
    <widget class="QFrame" name="footerFrame">
     <property name="minimumSize">
      <size>
       <width>0</width>
       <height>80</height>
      </size>
     </property>
     <property name="maximumSize">
      <size>
       <width>16777215</width>
       <height>80</height>
      </size>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: white; border-top: 2px solid #e0e0e0;</string>
     </property>
     <property name="frameShape">
      <enum>QFrame::Shape::StyledPanel</enum>
     </property>
     <layout class="QHBoxLayout" name="footerLayout">
      <property name="leftMargin">
       <number>30</number>
      </property>
      <property name="topMargin">
       <number>20</number>
      </property>
      <property name="rightMargin">
       <number>30</number>
      </property>
      <property name="bottomMargin">
       <number>20</number>
      </property>
      <item>
       <spacer name="footerSpacer">
        <property name="orientation">
         <enum>Qt::Orientation::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="buttonCancel">
        <property name="minimumSize">
         <size>
          <width>120</width>
          <height>45</height>
         </size>
        </property>
        <property name="cursor">
         <cursorShape>PointingHandCursor</cursorShape>
        </property>
        <property name="styleSheet">
         <string notr="true">QPushButton {
    background-color: #f8f9fa;
    color: #5a6c7d;
    border: 2px solid #e0e0e0;
    border-radius: 8px;
    font-size: 14px;
    font-weight: bold;
    padding: 10px 20px;
}

QPushButton:hover {
    border-color: #f5576c;
    color: #f5576c;
}

QPushButton:pressed {
    background-color: #e0e0e0; /* darker grey for pressed feedback */
    border-color: #f5576c;
    color: #f5576c;
}</string>
        </property>
        <property name="text">
         <string>Cancel</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="buttonSave">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="minimumSize">
         <size>
          <width>120</width>
          <height>45</height>
         </size>
        </property>
        <property name="cursor">
         <cursorShape>PointingHandCursor</cursorShape>
        </property>
        <property name="styleSheet">
         <string notr="true">QPushButton {
    background-color: #f5576c;
    color: #ffffff;
    border: none;
    border-radius: 8px;
    font-size: 14px;
    font-weight: bold;
    padding: 10px 20px;
    transition: background-color 150ms;
}
QPushButton:hover {
    background-color: #e5475c;
}
QPushButton:pressed {
    background-color: #d43f52;
}
QPushButton:disabled {
    background-color: #cccccc;  /* gray background for disabled */
    color: #666666;             /* darker gray text */
}</string>
        </property>
        <property name="text">
         <string>üíæ  Save</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
